spec/ics-002-client-semantics/README.md

```diff
@@ -468,28 +449,15 @@ function createClient(
     abortTransactionUnless(validateClientIdentifier(id))
     abortTransactionUnless(privateStore.get(clientStatePath(id)) === null)
     abortSystemUnless(provableStore.get(clientTypePath(id)) === null)
-    clientState = clientType.initialize(consensusState)
-    privateStore.set(clientStatePath(id), clientState)
+    clientType.initialise(consensusState)
     provableStore.set(clientTypePath(id), clientType)
 }
 ```
 
 #### Query
 
-Client consensus state and client internal state can be queried by identifier. The returned
-client state must fulfil an interface allowing membership / non-membership verification.
-
-```typescript
-function queryClientConsensusState(id: Identifier): ConsensusState {
-    return provableStore.get(consensusStatePath(id))
-}
-```
-
-```typescript
-function queryClient(id: Identifier): ClientState {
-    return privateStore.get(clientStatePath(id))
-}
-```
+Client consensus state and client internal state can be queried by identifier, but
+the specific paths which must be queried are defined by each client type.
```

Comment:
> the specific paths which must be queried are defined by each client type.

why is this now defined per client? All clients need to still be able to query the consensus and client states

Reply:
The client state, yes, but the consensus state, not necessarily, e.g. for a solo machine there is not a consensus state stored for each height. Client types can be required to expose `queryConsensusState` I suppose - is this required? The relayer should use a client-type-agnostic method.

Reply:
> The client state, yes, but the consensus state, not necessarily, e.g. for a solo machine there is not a consensus state stored for each height. Client types can be required to expose queryConsensusState I suppose...

👍 It'd be awesome if we could define a standard interface for all clients with the functions that they need to implement.

Reply:
Aye, agreed. I'll add that.

Reply:
With regards to a standard interface, I completely agree, but the only parts of that I can reliably guess are those which will be required by relayers (querying clients), so if you want additional functions for convenience or CLI UX, let me know.

Reply:
I defined a few interfaces for clients in cosmos/cosmos-sdk#5485

Reply:
Yeah. I think I'll split that out to a separate issue & we can discuss first. https://github.com/cosmos/ics/issues/356